awsutil/generate_credentials.go

```diff
@@ -379,3 +399,14 @@ func stsSigningResolver(service, region string, optFns ...func(*endpoints.Option
 	defaultEndpoint.SigningRegion = region
 	return defaultEndpoint, nil
 }
+
+// FetchTokenContents allows the use of the content of a token in the
+// WebIdentityProvider, instead of the path to a token. Useful with a
+// serviceaccount token requested directly from the EKS/K8s API, for example.
+type FetchTokenContents string
+
+var _ stscreds.TokenFetcher = (*FetchTokenContents)(nil)
+
+func (f FetchTokenContents) FetchToken(_ aws.Context) ([]byte, error) {
+	return []byte(f), nil
```

Comment:
Gotcha, and should `CredentialsConfig.WebIdentityToken` be a `[]byte` as well? 

For the operator case I believe we will have to cast from string to []byte somewhere along the line, since the k8s API returns the token as a string, but that's easily done on the operator side.

Reply:
I was thinking `WebIdentityToken` and its functional option can still be a `string`, and Go will accept passing it as `FetchTokenContents(c.WebIdentityToken)` without any further casting. No strong preference if you prefer consistently treating it as `[]byte` all the way through though.

Reply:
Oh gotcha, I like that better, updated in 3bde4d9